setup_routing.go

```diff
@@ -52,11 +52,12 @@ func setupDelegatedRouting(cfg Config, dnsCache *cachedDNS) ([]routing.Routing,
 	)
 
 	for _, endpoint := range cfg.RoutingV1Endpoints {
-		rv1Opts := []routingv1client.Option{routingv1client.WithHTTPClient(httpClient)}
-		if endpoint != cidContactEndpoint {
-			rv1Opts = append(rv1Opts, routingv1client.WithStreamResultsRequired())
-		}
-		delegatedRouter, err := delegatedHTTPContentRouter(endpoint, rv1Opts...)
+		delegatedRouter, err := delegatedHTTPContentRouter(endpoint,
+			routingv1client.WithHTTPClient(httpClient),
+			routingv1client.WithProtocolFilter(cfg.RoutingV1FilterProtocols), // IPIP-484
+			routingv1client.WithStreamResultsRequired(),                      // https://specs.ipfs.tech/routing/http-routing-v1/#streaming
```

Comment:
Out of scope for this PR, but...

According to the docs,
>server MAY respond with non-streaming application/json response even if the client requested streaming

So, would his option be better named `WithStreamResultsRequested` or `WithStreamResultsAccepted`?

Reply:
Yes, the name is confusing and should be adjusted. 
(afaik this config was onl yadded because cid.contact errored on streaming requests, but it was fixed since then)